components/exercises/chapter-5-oscar/Everything/WeatherApp.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { every } from './every';
+import './styles.css';
+
+export type WeatherData = {
+  name?: string;
+  main?: {
+    temp: number;
+    temp_min: number;
+    temp_max: number;
+    feels_like: number;
+    humidity: number;
+    pressure: string;
+  };
+  weather?: [{ description: string }];
+  wind?: {
+    speed: number;
+  };
+};
+
+const API_KEY = '297cdcb5306dab20d8338f39e1459828';
+
+/**
+ * Allows users to search for weather information based on a location input. It fetches data from the OpenWeatherMap API
+ * and displays it to the user. Also, compare the temperatures and provide a conditional message based on the temperature feeling.
+ * If an error occurs an error message is displayed.
+ * @returns - The WeatherApp component returns a user interface that
+ * allows users to input a location and view weather information.
+ */
+export default function WeatherApp(): JSX.Element {
+  const [weatherData, setWeatherData] = useState<WeatherData>({});
+  const [locationInput, setLocationInput] = useState<string>('');
+  const [error, setError] = useState<string | null>(null);
+
+  // Construct the API URL based on user input
+  const apiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${locationInput}&units=metric&appid=${API_KEY}`;
+
+  // Handle user's Enter key press to trigger weather data retrieval
+  async function handleFormSubmit(
+    event: React.FormEvent<HTMLFormElement>
+  ): Promise<void> {
+    event.preventDefault();
+    try {
+      // Fetch weather data from the API using Axios
+      const response = await axios.get<WeatherData>(apiUrl);
```

Comment:
Why axios? can't we just use `fetch`?

Reply:
Because I never have used it before and I thought it would be a good opportunity to try

Reply:
refactored, thank you